core/controllers/improvements.py

```diff
@@ -27,6 +27,31 @@
 from core.domain import exp_fetchers
 from core.domain import improvements_domain
 from core.domain import improvements_services
+from core.domain import user_services
+
+
+def get_task_dict_with_username_and_profile_picture(task_entry):
+    """Returns a task entry dict with the username and profile picture
+    URL inserted.
+
+    Args:
+        task_entry: improvements_domain.TaskEntry. The TaskEntry domain object.
+
+    Returns:
+        TaskEntryDict. TaskEntry dict username and profile picture
+        URL inserted.
+    """
+
+    task_entry_dict = task_entry.to_dict()
+    if task_entry.resolver_id:
```

Comment:
Why is this `if` needed?

Reply:
@vojtechjelinek This `if` is needed because the behaviour expected is to insert the username and profile picture if `resolver_id` is set, else return the existing dict.

Reply:
Okay, so resolver ID can also be `None`?

Reply:
Yes, it is `None` at some places in the code.
https://github.com/oppia/oppia/blob/741cc401ec375d4bd148dd547a718db7bcdb7499/core/domain/improvements_domain_test.py#L110

https://github.com/oppia/oppia/blob/5013f934da55868696924d4db60bde413eb2a616/core/domain/improvements_services_test.py#L47

Reply:
Can it be `None` in non-test files?

Reply:
I have the same question as Vojta here.